cpp/src/rolling/rolling.cu

```diff
@@ -623,5 +625,216 @@ std::unique_ptr<column> rolling_window(column_view const& input,
   }
 }
 
+std::unique_ptr<column> grouped_rolling_window(table_view const& grouping_keys,
+                                               column_view const& input,
+                                               size_type preceding_window,
+                                               size_type following_window,
+                                               size_type min_periods,
+                                               std::unique_ptr<aggregation> const& aggr,
+                                               rmm::mr::device_memory_resource* mr)
+{
+  if (input.size() == 0) return empty_like(input);
+
+  if (grouping_keys.num_columns() == 0) {
+    // No grouping_keys specified. Treat as non-grouped case.
+    return rolling_window(input, preceding_window, following_window, min_periods, aggr, mr);
+  }
+
+  CUDF_EXPECTS((grouping_keys.num_rows() == input.size()), 
+               "Size mismatch between grouping_keys and input vector.");
+
+  CUDF_EXPECTS((min_periods > 0), "min_periods must be positive");
+
+  using sort_groupby_helper = cudf::experimental::groupby::detail::sort::sort_groupby_helper;
+  sort_groupby_helper helper{grouping_keys, cudf::include_nulls::NO, cudf::sorted::YES};
```

Comment:
Just want to point out that sort helper will re-sort if it sees a particular argument combination. If you use `include_nulls::NO` and `sorted::YES` then you'd also have to ensure that the keys don't have nulls. 
https://github.com/rapidsai/cudf/blob/9bd1c2f280d61e545b18083b68ab33eec8134e61/cpp/include/cudf/detail/groupby/sort_helper.hpp#L72-L77
The reason is that we have to provide helper data assuming the nulls are not in the contiguous key data.
```
keys = [1, 1, 1, NULL, 2, 2]
```
becomes
```
keys = [1, 1, 1, 2, 2]
```

Reply:
Oh, thank you for pointing this out; I wouldn't have known. 

The keys come in pre-sorted. I should probably be setting `include_nulls::YES`.

Reply:
I just tested that this works.

Reply:
Hello, @devavret. Just confirming here that I've changed the parameters as you suggested, to avoid re-sorting the data.